### ISSUE TITLE
Add Ingest Module/Job to SleuthKit Case

### ISSUE THREAD

**Issue by mohammadsuhail98** [sleuthkit/sleuthkit#2939]
I am trying to add an ingest module to the sleuth kit case using the available methods which are **addIngestModule** and **addIngestJob** but given the source code I read, it only adds the ingest job to the case without any method of triggering or starting the ingest job.

<img width="715" alt="Screenshot 2024-08-12 at 12 21 12 PM" src="https://github.com/user-attachments/assets/8b73f58b-9b30-48c2-b3d5-0fa2bb6a75e6">

**Comment by APriestman** [comment]
Please see the documentation here on how to create your own ingest module. 
https://www.sleuthkit.org/autopsy/docs/api-docs/4.19.3/

Specifically:
https://www.sleuthkit.org/autopsy/docs/api-docs/4.19.3/mod_ingest_page.html

You should not be called addIngestJob or addIngestModule.

**Comment by mohammadsuhail98** [comment]
Yes but I am not looking for creating my own ingest module, I want to use an already existing ingest module as shown in the screenshot which is **"PictureAnalyzerIngestModule"**, I understand the use of the factory now but the usage process of the ingest module is not very clear for me, these are the steps that I understood:

1- create factory instance.
2- get the ingestModule instance using the factory method **createFileIngestModule()**;
3- use the **startUp()**, **process()** and **shutdown()** methods of that module to handle its life cycle.
4- here comes my problem, to trigger the **startUp()** method It requires a **jobContext** as a parameter, and to instantiate an instance of the jobContext it requires **IngestJobExecutor**, and I do not know how can I get an instance of the **IngestJobExecutor** so I can trigger the **startUp()** method.
